lib/trento/users.ex

```diff
@@ -0,0 +1,78 @@
+defmodule Trento.Users do
+  @moduledoc """
+  The Users context.
+  """
+
+  use Pow.Ecto.Context,
+    repo: Trento.Repo,
+    user: Trento.Users.User
+
+  import Ecto.Query, warn: false
+  alias Trento.Repo
+
+  alias Trento.Users.User
+
+  @impl true
+  @doc """
+    get_by function overrides the one defined in Pow.Ecto.Context,
+    we retrieve the user by username as traditional Pow flow but we also exclude
+    deleted and locked users
+  """
+  def get_by(clauses) do
+    username = clauses[:username]
+
+    User
+    |> where([u], is_nil(u.deleted_at) and is_nil(u.locked_at) and u.username == ^username)
+    |> Repo.one()
+  end
+
+  def list_users do
+    User
+    |> where([u], is_nil(u.deleted_at))
+    |> Repo.all()
+  end
+
+  def get_user(id) do
+    case User
+         |> where([u], is_nil(u.deleted_at) and u.id == ^id)
+         |> Repo.one() do
+      nil -> {:error, :not_found}
+      user -> {:ok, user}
+    end
+  end
+
+  def create_user(attrs \\ %{}) do
+    %User{}
+    |> User.changeset(attrs)
+    |> Repo.insert()
+  end
+
+  def update_user(%User{id: 1}, _), do: {:error, :operation_not_permitted}
```

Comment:
Maybe a comment that this is the admin user which must not be deleted

Reply:
it's not possible to update the admin user, we have environment variables to control the attributes, it will overwrite things in the release task